File: shoring/desk/src/main/java/com/panopset/marin/oldswpkg/games/blackjack/LayoutPlayer.java
package com.panopset.marin.oldswpkg.games.blackjack;

import com.panopset.compat.Tile;
import com.panopset.blackjackEngine.CycleSnapshot;
import com.panopset.blackjackEngine.HandPlayer;
import com.panopset.blackjackEngine.Player;

public class LayoutPlayer extends Layout {
    public final int chipW;
    public final int chipH;

    public LayoutPlayer(Tile tile, int[] cardDim, CycleSnapshot cs, int[] chipDim) {
        super(tile, cardDim, cs);
        chipW = chipDim[0];
        chipH = chipDim[1];
    }

    private Integer playerChipY;

    public Integer getPlayerChipY() {
        if (playerChipY == null) {
            playerChipY = t.getTop() + cardH;
        }
        return playerChipY;
    }

    private Integer playerMsgY;

    public Integer getPlayerMsgY() {
        if (playerMsgY == null) {
            playerMsgY = getPlayerChipY() + (chipH * 2);
        }
        return playerMsgY;
    }

    private Integer playerCardXstart;

    public Integer getPlayerCardXstart() {
        if (playerCardXstart == null) {
            int cardsWidth = 0;
            for (Player p : cs.getPlayers()) {
                for (HandPlayer h : p.getHands()) {
                    if (!h.cards.isEmpty()) {
                        cardsWidth = cardsWidth + cardW;
                        cardsWidth = cardsWidth + (cardSpacer * h.cards.size());
                    }
                }
            }
            playerCardXstart = centerW - (cardsWidth / 2);
        }
        return playerCardXstart;
    }

    private Integer playerY;

    public Integer getPlayerY() {
        if (playerY == null) {
            playerY = t.getTop() + cardH;
        }
        return playerY;
    }
}

File: code/shoring/desk/src/main/kotlin/com/panopset/marin/oldswpkg/games/blackjack/LayoutPlayer.kt
package com.panopset.marin.oldswpkg.games.blackjack

import com.panopset.blackjackEngine.BlackjackGameState
import com.panopset.compat.Tile

class LayoutPlayer(tile: Tile?, cardDim: IntArray?, cs: BlackjackGameState?, chipDim: IntArray) :
    Layout(tile!!, cardDim!!, cs!!) {
    val chipW: Int
    val chipH: Int
    var playerChipY: Int? = null
        get() {
            if (field == null) {
                field = t.top + cardH
            }
            return field
        }
        private set
    var playerMsgY: Int? = null
        get() {
            if (field == null) {
                field = playerChipY!! + chipH * 2
            }
            return field
        }
        private set
    var playerCardXstart: Int? = null
        get() {
            if (field == null) {
                var cardsWidth = 0
                for (p in cs.players) {
                    for (h in p.hands) {
                        if (h.getBlackjackCards().isNotEmpty()) {
                            cardsWidth += cardW
                            cardsWidth += cardSpacer * h.getBlackjackCards().size
                        }
                    }
                }
                field = centerW - cardsWidth / 2
            }
            return field
        }
        private set
    var playerY: Int? = null
        get() {
            if (field == null) {
                field = t.top + cardH
            }
            return field
        }
        private set

    init {
        chipW = chipDim[0]
        chipH = chipDim[1]
    }
}
